src/adaptors/articles.ts

```diff
@@ -0,0 +1,13 @@
+import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
+
+export interface Articles extends Iterable<JsonLdObj> {
+    count(): Promise<number>;
```

Comment:
Feels a bit like this should be `readonly length: Promise<number>` instead.

Reply:
That might be confusing mixing with the `Array { length: number }` property.

Reply:
Was thinking about them in the same way, but the promise is different...